src/native/fs/fs_napi.cpp

```diff
@@ -1351,6 +1352,47 @@ struct GetPwName : public FSWorker
     }
 };
 
+/**
+ * Utimensat is an fs op
+ * change file times with nanosecond precision (access and modification times)
+ * see https://man7.org/linux/man-pages/man3/utimensat.3p.html
+ */
+struct Utimensat : public FSWorker
+{
+    std::string _path;
+    struct timespec _times[2];
+    Utimensat(const Napi::CallbackInfo& info)
+        : FSWorker(info)
+    {
+        _path = info[1].As<Napi::String>();
+        Napi::Object napi_times = info[2].As<Napi::Object>();
+        std::string new_atime;
+        get_time_from_info(napi_times, "actime", _times[0], new_atime);
+        std::string new_mtime;
+        get_time_from_info(napi_times, "modtime", _times[1], new_mtime);
+        Begin(XSTR() << "utimensat " << DVAL(_path) << DVAL(new_atime) << DVAL(new_mtime));
+    }
+    virtual void Work()
+    {
+        SYSCALL_OR_RETURN(utimensat(AT_FDCWD, _path.c_str(), _times, 0));
```

Comment:
does mac have utimensat() function and same behavior?

Reply:
please always make sure you are not breaking native build for the developers on mac, I sent you on slack the man utimensat

Reply:
right, than it looks like it works on mac. also I should have mentioned it before: utimensat is defined in POSIX, so it should work on mac the same. even though you can never know. We will test this on mac regardless before pushing